augur/filter.py

```diff
@@ -310,6 +312,20 @@ def run(args):
             if args.priority: # read priorities
                 priorities = read_priority_scores(args.priority)
 
+            if spg is None:
+                # this is only possible if we have imposed a maximum number of samples
+                # to produce.  we need binary search until we have the correct spg.
+                try:
+                    spg = _calculate_spg(
+                        args.subsample_max_sequences,
+                        [len(sequences_in_group)
+                         for sequences_in_group in seq_names_by_group.values()],
+                    )
+                except TooManyGroupsError as ex:
+                    print(f"ERROR: {ex.args[0]}", file=sys.stderr)
```

Comment:
Instead of printing the message via the args attribute, could you use the string representation of the `ex` object? This produces the same result but is slightly easier to read. For example:

```python
print(f"ERROR: {ex}", file=sys.stderr)
```

Reply:
This doesn't work as stringifying an exception by default returns the repr of the arguments.  I will make it explicit by defining a __str__ method.

Reply:
That's odd...I originally had the same expectation, but when I run the example code I wrote above, it works as expected. It looks like [f-strings call the `__format__` method of each expression](https://www.python.org/dev/peps/pep-0498/#specification). The default behavior of `__format__` [appears to be to apply `str()` to the object](https://www.python.org/dev/peps/pep-3101/#controlling-formatting-on-a-per-type-basis). I couldn't find a place where `ValueError` defines its own `__format__` method though.

Here is a test I ran locally to convince myself though:

```
>>> exception = ValueError("this failed")
>>> repr(exception)
"ValueError('this failed',)"
>>> str(exception)
'this failed'
>>> print(exception)
this failed
>>> print(f"The exception was: {exception}")
The exception was: this failed
```

Do you have an example of how to get the exception to return its `__repr__` instead of its `__str__`? It doesn't matter for this PR, but I'm just curious for my own Python edification.

Reply:
You're right, it's calling "__str__", but I think that resolves to repr(*args).

```
In [1]: class ABC(ValueError): 
   ...:     pass 
   ...:                                                                                                                                              

In [2]: try: 
   ...:     raise ABC("abc", "def") 
   ...: except ABC as ex: 
   ...:     print(f"{ex}") 
   ...:                                                                                                                                              
('abc', 'def')
```

I want it to be the first argument explicitly.

Reply:
That totally makes sense. Thanks!